core/sis-utility/src/main/java/org/apache/sis/util/iso/DefaultScopedName.java

```diff
@@ -293,10 +293,10 @@ public synchronized GenericName tail() {
         if (tail == null) {
             final int size = parsedNames.size();
             switch (size) {
-                default: tail = new DefaultScopedName(parsedNames.subList(1, size)); break;
                 case 2:  tail = parsedNames.get(1); break;
                 case 1:  // fall through
                 case 0:  throw new AssertionError(size);
+                default: tail = new DefaultScopedName(parsedNames.subList(1, size)); break;
```

Comment:
The previous code had a logical ordering from highest value (where `default` stands for all values greater than 2) to zero.

Reply:
I considered it to be a small improvement due to https://sonarcloud.io/organizations/default/rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck

Going to undo.

Reply:
In this switch the default is not for defensing programming, but is a case expected to be run in normal execution (actually most of the time for that switch statement).